Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -1504,11 +1514,13 @@ extension PurchasesOrchestrator {
     func syncPurchases(receiptRefreshPolicy: ReceiptRefreshPolicy,
                        isRestore: Bool,
                        initiationSource: ProductRequestData.InitiationSource) async throws -> CustomerInfo {
-        return try await Async.call { completion in
+        return try await withCheckedThrowingContinuation { continuation in
```

Comment:
this seems simpler. Wonder if we should have an alternative to withCheckedThrowingContinuation that always resumes with an assert on main thread